lib/PlatoonService/src/CollisionAvoidance.h

```diff
@@ -53,75 +53,78 @@
  * Types and Classes
  *****************************************************************************/
 
-/** Follower class. */
-class Follower
+/**
+ * Collision avoidance class.
+ */
+class CollisionAvoidance
 {
 public:
     /**
-     * Follower Constructor.
-     */
-    Follower() : m_lastHeartbeatTimestamp(0U), m_status(0U)
-    {
-    }
-
-    /**
-     * Follower Destructor.
-     */
-    ~Follower()
-    {
-    }
-
-    /**
-     * Get the last heartbeat timestamp.
+     * Collision avoidance constructor
      *
-     * @return Last heartbeat timestamp.
+     * @param[in] closestProximityRangeValue Value equivalent to the closest proximity range that can be measured.
+     * @param[in] rangeThreshold Range at which the vehicle should stop to avoid collision.
      */
-    uint32_t getLastHeartbeatTimestamp() const
-    {
-        return m_lastHeartbeatTimestamp;
-    }
+    CollisionAvoidance(uint8_t closestProximityRangeValue, uint8_t rangeThreshold) :
+        m_closestProximityRangeValue(closestProximityRangeValue),
+        m_rangeThreshold(rangeThreshold){};
 
     /**
-     * Set the last heartbeat timestamp.
-     *
-     * @param[in] timestamp    Last heartbeat timestamp.
+     * Default Destructor.
      */
-    void setLastHeartbeatTimestamp(uint32_t timestamp)
+    ~CollisionAvoidance()
     {
-        m_lastHeartbeatTimestamp = timestamp;
     }
 
     /**
-     * Get the status.
+     * Limit the speed of the vehicle to avoid collision.
      *
-     * @return Status.
+     * @param[out] speedSetpoint The speed setpoint to be limited.
+     * @param[in] vehicleData The vehicle data.
      */
-    uint8_t getStatus() const
+    void limitSpeedToAvoidCollision(int16_t& speedSetpoint, const Telemetry& vehicleData) const
     {
-        return m_status;
+        /* Is vehicle is closer than the threshold? */
+        if (vehicleData.proximity >= m_rangeThreshold)
+        {
+            speedSetpoint = 0;
+        }
     }
 
     /**
-     * Set the status.
+     * Limit the speed of the vehicle to avoid collision.
      *
-     * @param[in] status    Status.
+     * @param[out] leftSpeedSetpoint The left motor speed setpoint to be limited.
+     * @param[out] rightSpeedSetpoint The right motor speed setpoint to be limited.
+     * @param[in] vehicleData The vehicle data.
      */
-    void setStatus(uint8_t status)
+    void limitSpeedToAvoidCollision(int16_t& leftSpeedSetpoint, int16_t& rightSpeedSetpoint,
+                                    const Telemetry& vehicleData) const
     {
-        m_status = status;
+        /* Is vehicle is closer than the threshold? */
+        if (vehicleData.proximity >= m_rangeThreshold)
+        {
+            leftSpeedSetpoint  = 0;
+            rightSpeedSetpoint = 0;
+        }
     }
 
 private:
-    /** Last heartbeat timestamp. */
-    uint32_t m_lastHeartbeatTimestamp;
+    /** Closest proximity range value. */
+    uint8_t m_closestProximityRangeValue;

```

Comment:
unit?

Reply:
IMHO there was a enumeration about, isn't it?

Reply:
Yes there is an enum for this, but it is in SerialMuxChannels.h in the application-specific folder. I will look for a solution that does not involve code (enum) duplication.